analyzer/codechecker_analyzer/analyzer_context.py

```diff
@@ -53,6 +53,115 @@ def __len__(self):
         return len(self.store)
 
 
+class ProfileMap(Mapping):
+    """
+    A dictionary which maps checker names and checker groups to profile names.
+    A checker or checker group may be the member of multiple profiles.
+    """
+    def __init__(self, profile_map_file):
+        self.store = load_json_or_empty(profile_map_file, {})
+        self.__check_json_format(profile_map_file)
+
+    def __check_json_format(self, profile_map_file):
+        """
+        This function checks the format of the given profile map config file.
+        If this file doesn't meet the requirements, then CodeChecker exits with
+        status 1.
+
+        {
+          "available_profiles": {
+            "profile1": "description1",
+            "profile2": "description2"
+          },
+          "analyzers": {
+            "analyzer1": {
+              "profile1": ['checker1', 'checker2']
+            },
+            "analyzer2": {
+              "profile1": ['checker3'],
+              "profile2": ['checker3', 'checker4']
+            }
+          }
+        }
+        """
+        if 'available_profiles' not in self.store:
+            LOG.error('Format error in %s: "available_profiles" key not '
+                      'found', profile_map_file)
+            sys.exit(1)
```

Comment:
I think it would be better to exit only on the top main files not here it is much harder to write test for this function.
An error/exception could be returned and handled if there is a problem with the format.
Returning the error like that the profile_map_file variable would not be needed here (it is used only to print out the error).